### PR TITLE
Freebsd Install

### PR DESCRIPTION
**Pull request by No9**
This is a patch to enable llnode to work on freebsd from an npm install. 
I think most of the code and docs are pretty straight forward.
It imitates the linux approach currently in place. 
I have intentionally not modified the linux path through the code to make it easy to reason about. 

I am not to sure about the package.json line as it looks verbose 
https://github.com/nodejs/llnode/compare/master...No9:freebsd-install?expand=1#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R12 
Maybe this should go in its own script now?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
   "//": "(Blame C++)",
   "scripts": {
     "preinstall": "node scripts/configure.js",
-    "install": "./gyp_llnode && make -C out/",
+    "install": "./gyp_llnode && if [ `uname` == 'FreeBSD' ]; then gmake -C out/; else make -C out/; fi",
```
Comment (indutny) [inline package.json]: `gmake -C out/ || make -C out/`? :)

**Comment by No9** [comment]
Thanks @indutny I knew there was a smell there :) 
Updated the PR with your line and tested on FreeBSD 11 and Ubuntu 16.04

File: README.md [hunk 2]
```diff
@@ -75,6 +75,31 @@ make -C out/ -j9
...
+
+# Configure
+./gyp_llnode -Dlldb_dir=/usr/local/llvm39/
+
+# Build
+gmake -C out/ -j9
+
+# Install
+ComputeSystemPluginsDirectory is not implemented in FreeBSD
+The llnode.so will have to be loaded manually
```
Comment (bnoordhuis) [inline README.md]: This paragraph should presumably be outside the triple backticks?  Aside: can you punctuate sentences?

File: package.json [hunk 3]
```diff
@@ -9,7 +9,7 @@
   "//": "(Blame C++)",
   "scripts": {
     "preinstall": "node scripts/configure.js",
-    "install": "./gyp_llnode && make -C out/",
+    "install": "./gyp_llnode && gmake -C out/ || make -C out/",
```
Comment (bnoordhuis) [inline package.json]: Logic bug: this is parsed as `(./gyp_llnode && gmake -C out/) || make -C out/` which makes `make -C out/` run even when gyp_llnode fails.  I speculate the pile up of errors is going to look confusing to users so I'd suggest adding parentheses.

File: scripts/configure.js [hunk 4]
```diff
@@ -61,6 +61,27 @@ if (osName === 'Darwin') {
...
+    console.log('Unable to locate lldb binary. llnode installation failed.');
+    process.exit(1);  
+  }
+  
+  var installedHeadersDir = getFreeBSDHeadersDir(lldbVersion);
+  if (installedHeadersDir === undefined) {
+    // As this is a BSD we know this system is in an improper state
+    // So we can exit with an error
+    console.log('The system isn\'t set up correcly.'); 
+    console.log('Try `pkg install llvm39` and `ln -s /usr/local/bin/lldb39 /usr/bin/lldb`');
```
Comment (bnoordhuis) [inline scripts/configure.js]: Long line, please wrap it at 80 columns.

File: package.json (hunk 3 above)
Comment (gibfahn) [inline package.json]: Maybe:

```bash
"{ ./gyp_llnode && gmake -C out/; } || make -C out/"
```

File: README.md (hunk 2 above)
Comment (hhellyer) [inline README.md]: I've redone the README.md changes as in the meantime one of my changes re-organised it so these didn't merge. I'll sort out the merge for this today.

**Comment by hhellyer** [comment]
I've pulled the README.md changes into the new format and persuaded github this can merge. @bnoordhuis - Are you happy for this to merge now?

@No9 (or anyone else with a FreeBSD system) - Does npm -g install now work on FreeBSD from this branch? I think it should but it would be nice to confirm before I bump the version and push a new level to npmjs.org to get that available to FreeBSD users.

**Comment by No9** [comment]
@hhellyer tested and made a change because of a possible environment config issue. See notes https://github.com/nodejs/llnode/pull/96/commits/a6641420834565bd8d9e44f81bbee69fd2a88fbd 
I've validated this on FreeBSD 11 and Ubuntu 16.04 but would another once over would be good.

You mentioned a linter in one of your PRs. 
Am I missing a validation step so I can reduce the nits in my PRs?

**Comment by hhellyer** [comment]
@No9 - No, I just run a linter locally to reduce nits in my own PR's.
I've tried installing from your branch on a Mac and it works with your latest change.

**Comment by hhellyer** [comment]
LGTM - Ben's comments have been addressed.

**Pull request merged**